tests_integration/step_prepare.py

```diff
@@ -0,0 +1,75 @@
+"""
+After closing all open Pull Requests in the conda-forge staging organization,
+runs the prepare() function of all test cases of the current test scenario to prepare the test environment.
+
+Expects the scenario ID to be present in the environment variable named SCENARIO_ID.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+from github import Github
+
+from tests_integration.collect_test_scenarios import get_test_scenario
+from tests_integration.lib.integration_test_helper import IntegrationTestHelper
+from tests_integration.lib.shared import (
+    ENV_TEST_SCENARIO_ID,
+    FEEDSTOCK_SUFFIX,
+    GitHubAccount,
+    get_github_token,
+    get_test_case_modules,
+    setup_logging,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def close_all_open_pull_requests():
+    github = Github(get_github_token(GitHubAccount.CONDA_FORGE_ORG))
+    org = github.get_organization(GitHubAccount.CONDA_FORGE_ORG)
+
+    for repo in org.get_repos():
+        if not repo.name.endswith(FEEDSTOCK_SUFFIX):
+            continue
+        for pr in repo.get_pulls(state="open"):
+            pr.create_issue_comment(
+                "Closing this PR because it is a leftover from a previous test run."
+            )
+            pr.edit(state="closed")
+
+
+def reset_cf_graph():
+    IntegrationTestHelper.overwrite_github_repository(
+        GitHubAccount.REGRO_ORG,
+        "cf-graph-countyfair",
+        Path(__file__).parent / "resources" / "empty-graph",
+        branch="master",
+    )
+
+
+def run_all_prepare_functions(scenario: dict[str, str]):
+    test_helper = IntegrationTestHelper()
+    for test_module in get_test_case_modules(scenario):
+        try:
+            logging.info("Preparing %s...", test_module.__name__)
+            test_module.prepare(test_helper)
+        except AttributeError:
+            raise AttributeError("The test case must define a prepare() function.")
+
+
+def main(scenario_id: int):
+    close_all_open_pull_requests()
+    reset_cf_graph()
+
+    scenario = get_test_scenario(scenario_id)
+
+    logging.info("Preparing test scenario %d...", scenario_id)
+    logging.info("Scenario: %s", scenario)
+
+    run_all_prepare_functions(scenario)
+
+
+if __name__ == "__main__":
+    setup_logging(logging.INFO)
```

Comment:
it could be nice to set this to debug when debug logging is enabled in the run: you can test this by checking `$RUNNER_DEBUG`: https://github.com/actions/toolkit/blob/5e8c25d1f5d92a53f2438de763f92d4809dd666f/packages/core/src/core.ts#L234